File: back-end/reverie/src/main/java/com/sapienza/reverie/Repository/CharmRepository.java
package com.sapienza.reverie.Repository;

import com.sapienza.reverie.Model.Charm;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface CharmRepository extends JpaRepository<Charm,Long> {

}

File: app/app/src/main/java/com/sapienza/reverie/domain/repository/CharmRepository.kt
package com.sapienza.reverie.domain.repository

import com.sapienza.reverie.domain.model.CharmModel
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface CharmRepository {
    @GET("charm/created")
    suspend fun getCreatedCharms(@Query("user_id")userId : Long): List<CharmModel>

    @GET("charm/{id}")
    suspend fun getCharmById(@Path("id") id: Long): CharmModel

    @GET("charm/random")
    suspend fun getCarouselCharms(@Query("user_id") id: Long): List<CharmModel>

}